src/tiny_rsa/mod.rs

```diff
@@ -0,0 +1,164 @@
+//! Tiny RSA implementation
+//! TinyRSA module for educational cryptographic implementations.
+//! RSA Security Assumptions:
+//! - The security of RSA relies on the difficulty of factoring large integers.
+//! - The implementation is secure if the modulus is the product of two large random primes (they
+//!   are not random or big in these tests).
+//! - The size of the modulus should be at least 2048 bits.
+use std::{
+  iter::{Product, Sum},
+  ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Rem, Sub, SubAssign},
+};
+
+pub mod arithmetic;
+#[cfg(test)] mod tests;
+
+use crate::field::FiniteField;
+
+/// A field element for RSA operations with a modulus of P.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Default, PartialOrd)]
+pub struct RSAField<const N: usize> {
+  /// The value of the field element
+  pub value: usize,
+}
+
+/// Constructs a new RSAField element
+impl<const N: usize> RSAField<N> {
+  /// Constructs a new RSAField element
+  pub fn new(value: usize) -> Self {
+    let p = random_prime(2);
+    let q = random_prime(p);
+    assert_ne!(p, q);
+    Self { value: value % (p * q) }
+  }
+}
+
+impl<const N: usize> FiniteField for RSAField<N> {
+  const ONE: Self = Self { value: 1 };
+  const ORDER: usize = N;
+  // placeholder, fields without prime order don't have a primitive element
```

Comment:
I'm confused by this a bit. What is an example of an RSA field that doesn't have a primitive element? If it isn't prime order, it can't be a finite field.

Reply:
In reading a bit more about RSA, you are doing modulo `pq` for two primes `p` and `q`. So I don't think you're working over a finite field in this case, just a cyclic group.

Reply:
> I'm confused by this a bit. What is an example of an RSA field that doesn't have a primitive element? If it isn't prime order, it can't be a finite field.

Yes that makes sense to me. I think i could try to finagle a little bit better of an abstraction for it. I was tinkering with the field traits we have, but they didn't feel quite right.

Reply:
I don't know if you really want to work with this as a field since it is not a field